### ISSUE TITLE
Client crashing when entering Deep Dark Dimension on server

### ISSUE THREAD

**Issue by GameBurrow** [ItsDizzy/DiscordRP#2]
Original bug report & logfile: https://minecraft.curseforge.com/projects/the-society-transportcraft/issues/8

> Two attempts by two players. Both resulting in crash when entering deep dark after 'swirling' affects.

By looking at the logfile, it appears that client crashing has something to do with DiscordRP.
```
java.lang.NullPointerException: Unexpected error
	at java.util.Objects.requireNonNull(Objects.java:203)
	at java.util.Optional.<init>(Optional.java:96)
	at java.util.Optional.of(Optional.java:108)
	at me.paulhobbel.discordrp.api.impl.Registry.getDimension(Registry.java:25)
	at me.paulhobbel.discordrp.client.handlers.GameHandler.onJoinWorld(GameHandler.java:36)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_32_GameHandler_onJoinWorld_EntityJoinWorldEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraft.world.World.func_72897_h(World.java:3543)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1819)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1097)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

**Comment by paulhobbel** [comment]
Could you tell me which version of DiscordRP you have installed?

**Comment by GameBurrow** [comment]
```
| UCHIJA | discordrp  | 1.2.0.13 | DiscordRP-1.12.2-1.2.0.13.jar | None |
```
Full log file, with full list of mods was in the curse issue: https://media.forgecdn.net/attachments/235/642/crash-2018-08-23_00.txt

**Comment by paulhobbel** [comment]
Thank you, will look into it when I get home. It probably tries to load a dimension that does not exist. Normally it should fallback to a default dimension

**Comment by paulhobbel** [comment]
It's weird because this line should prevent it: https://github.com/ItsDizzy/DiscordRP/blob/e37f17bd0a3a9e6eb3a032dd89cd1f6f142aaff5/src/main/java/me/paulhobbel/discordrp/client/handlers/GameHandler.java#L36

**Comment by paulhobbel** [comment]
Could you test this version: https://ci.appveyor.com/api/buildjobs/hpdhsa4wewt8lhlv/artifacts/build%2Flibs%2FDiscordRP-1.12.2-1.2.16-dev.jar

**Comment by GameBurrow** [comment]
That fixes the issue.

**Comment by paulhobbel** [comment]
Will push this to an official release once I fixed some other less common issues

**Comment by GameBurrow** [comment]
Any chance seeing that official release soon?

**Comment by ghost** [comment]
@paulhobbel Can you please push this? This is kinda crucial.

**Comment by ghost** [comment]
you should push this im having the same issue with twilight forest

**Comment by paulhobbel** [comment]
I released a new versio v1.2.19 where this bug should be fixed
